provisioning/main.c

```diff
@@ -0,0 +1,345 @@
+/*
+ * Copyright (c) 2019, Arm Limited and affiliates
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#if defined(ATCA_HAL_I2C)
+#include "mbedtls/pk.h"
+#include "mbedtls/x509.h"
+#include "mbedtls/x509_csr.h"
+#include "psa/crypto.h"
+#include "psa/lifecycle.h"
+#include "atecc608a_se.h"
+#include "mbed_assert.h"
+
+/* The slot number for the device private key stored in the secure element by
+ * the secure element factory.
+ *
+ * ATECC608A secure elements with pre-provisioned keys often have that
+ * key present in slot 0, so we use 0 here for our example.
+ *
+ * Note that unlike most numerical values in the PSA Crypto API, 0 is a valid
+ * value for a slot number. Like many other secure elements, the ATECC608A
+ * numbers its slots from 0.
+ */
+#define EXAMPLE_FACTORY_KEY_SE_SLOT 0
+
+/* The slot number for the device private key which is generated within the
+ * secure element (never leaving the secure element) by this example
+ * provisioning application. */
+#define EXAMPLE_GENERATED_KEY_SE_SLOT 2
+
+/* The application-specific key ID for the secure element factory-provided
+ * device private key. This provisioning example application will associate the
+ * factory-provided key with this key ID for use by other applications. Any
+ * valid ID can be chosen here; the chosen ID does not need to correlate in any
+ * way with the physical location of the key (within the secure element). */
+#define EXAMPLE_FACTORY_KEY_ID 16
+
+/* The application-specific key ID for the device private key imported into the
+ * secure element by this example provisioning application. */
+#define EXAMPLE_GENERATED_KEY_ID 18
+
+/* Mbed TLS needs a CSPRNG to generate a CSR. Provide it a callback which uses
```

Comment:
```suggestion
/* Mbed TLS needs a random generator to generate a CSR. Provide it a callback which uses
```
The random generator is used for a random challenge and for blinding. A PRNG would not work for either use.